### ISSUE TITLE
Scroll restoration

### ISSUE THREAD

**Issue by kristiandupont** [kyeotic/raviger#111]
I currently use this extended version of `useRoutes`: 

```typescript

type ScrollRouteOptionParams = RouteOptionParams & {
  scrollRef: RefObject<HTMLElement> | 'body';
};

const useRoutesWithScroll = <T extends string>(
  routes: Routes<T>,
  { scrollRef, ...options }: ScrollRouteOptionParams
): JSX.Element => {
  const result = useRoutes(routes, options);

  const scrollToTop = useCallback(() => {
    if (scrollRef === 'body') {
      window.scrollTo(0, 0);
    } else {
      scrollRef.current.scrollTo(0, 0);
    }
  }, [scrollRef]);

  useEffect(scrollToTop, [result, scrollToTop]);

  return result;
};

```

It kinda works, but the true implementation would be able to restore scroll positions. I think it should be easy enough, but since you emit a `popstate` event when navigating (https://github.com/kyeotic/raviger/blob/main/src/navigate.ts#L73), it's hard to tell the difference between naviagtion and back-navigation. Could you use a different event or maybe tag the event somehow, so it's easy to filter out?

**Comment by kyeotic** [comment]
I'm open to doing this, but I'm not exactly sure how. What do you mean by "tag the event"?

**Comment by kristiandupont** [comment]
Sorry, holidays got in the way :-)
I am thinking something that would allow me to recognize them, like:
```typescript
const e = new PopStateEvent('popstate');
(e as any).__tag = 'synthetic';
dispatchEvent(e);
```
That way, I could check for this `__tag` and decide to ignore the event.

**Comment by kyeotic** [comment]
This was released under `3.1.0`
